iceoryx_utils/test/moduletests/test_cxx_forward_list.cpp

```diff
@@ -0,0 +1,1931 @@
+    EXPECT_THAT((++iter)->m_value, Eq(1584122));
+    EXPECT_THAT((++iter)->m_value, Eq(C_TOR_TEST_VALUE_DEFAULT_VALUE));
+    EXPECT_THAT((++iter)->m_value, Eq(C_TOR_TEST_VALUE_DEFAULT_VALUE));
+    EXPECT_THAT((++iter)->m_value, Eq(15842));
+}
+
+TEST_F(forward_list_test, RemoveIfOnetoEmptyList)
+{
+    forward_list<CTorTest, 10> cut1;
+    cut1.emplace_front(15842);
+
+    auto cnt = cut1.remove_if([](const CTorTest& cut) { return cut.m_value == 15842; });
+
+    EXPECT_THAT(cTor, Eq(0));
+    EXPECT_THAT(customCTor, Eq(1));
+    EXPECT_THAT(dTor, Eq(1));
+    EXPECT_THAT(cut1.size(), Eq(0));
+    EXPECT_THAT(cnt, Eq(1));
+}
+
+TEST_F(forward_list_test, RemoveIfWithFewMatches)
+{
+    forward_list<CTorTest, 10> cut1;
+    cut1.emplace_front(15842);
+    cut1.emplace_front();
+    cut1.emplace_front();
+
+    auto cnt = cut1.remove_if([](const CTorTest& cut) { return cut.m_value == C_TOR_TEST_VALUE_DEFAULT_VALUE; });
+
+    EXPECT_THAT(cTor, Eq(2));
+    EXPECT_THAT(customCTor, Eq(1));
+    EXPECT_THAT(dTor, Eq(2));
+    EXPECT_THAT(cut1.size(), Eq(1));
+    EXPECT_THAT(cnt, Eq(2));
+}
+
+TEST_F(forward_list_test, RemoveIfWithAllMatches)
+{
+    forward_list<CTorTest, 10> cut1;
+    cut1.emplace_front();
+    cut1.emplace_front();
+
+    auto cnt = cut1.remove_if([](const CTorTest& cut) { return cut.m_value == C_TOR_TEST_VALUE_DEFAULT_VALUE; });
+
+    EXPECT_THAT(cTor, Eq(2));
+    EXPECT_THAT(customCTor, Eq(0));
+    EXPECT_THAT(dTor, Eq(2));
+    EXPECT_THAT(cut1.size(), Eq(0));
+    EXPECT_THAT(cnt, Eq(2));
+}
+
+TEST_F(forward_list_test, RemoveIfAllFromList)
+{
+    forward_list<CTorTest, 10> cut1;
+    cut1.emplace_front(15842);
+    cut1.emplace_front();
+    cut1.emplace_front();
+
+    auto cnt = cut1.remove_if([](const CTorTest& cut) { return cut.m_value == 15842; });
+    cnt += cut1.remove_if([](const CTorTest& cut) { return cut.m_value == C_TOR_TEST_VALUE_DEFAULT_VALUE; });
+
+    EXPECT_THAT(cTor, Eq(2));
+    EXPECT_THAT(customCTor, Eq(1));
+    EXPECT_THAT(dTor, Eq(3));
+    EXPECT_THAT(cut1.size(), Eq(0));
+    EXPECT_THAT(cnt, Eq(3));
+}
+
+TEST_F(forward_list_test, writeContentViaDereferencedIterator)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        sut.emplace_front((const uint64_t)i);
+    }
+
+    auto cut{sut};
+    auto iter = cut.begin();
+    CTorTest element{TESTLISTCAPACITY + 1};
+    *iter = element;
+    EXPECT_THAT(cut.front().m_value, Eq(TESTLISTCAPACITY + 1));
+}
+
+TEST_F(forward_list_test, invalidIteratorErase)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        sut.emplace_front((const uint64_t)i);
+    }
+
+    auto iter = sut.begin();
+    sut.pop_front();
+
+    EXPECT_DEATH(sut.erase_after(iter), "");
+}
+
+TEST_F(forward_list_test, invalidIteratorIncrement)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        sut.emplace_front((const uint64_t)i);
+    }
+
+    auto iter = sut.cbegin();
+    sut.pop_front();
+
+    EXPECT_DEATH(++iter, "");
```

Comment:
Ok, trying to increment begin() iterator which is also end() since the list is empty. Should never happen in reasonable code with end() checks so is allowed to terminate.